Java/core/src/main/java/com/amazon/randomcutforest/tree/CompactRandomCutTreeDouble.java

```diff
@@ -111,26 +82,33 @@ protected boolean leftOf(double[] point, int dimension, double val) {
         return Arrays.copyOf(internal, internal.length);
     }
 
-    @Override
-    double[] getPointFromLeafNode(Integer node) {
-        return pointStore.get(nodeManager.getPointIndex(node));
+    double[] getPointSum(int ref, double[] point) {
+        if (nodeManager.isLeaf(ref)) {
+            if (getMass(ref) == 1) {
+                return getPointFromLeafNode(ref);
+            } else {
+                double[] answer = Arrays.copyOf(getPointFromLeafNode(ref), point.length);
+                for (int i = 0; i < point.length; i++) {
+                    answer[i] *= getMass(ref);
+                }
+                return answer;
+            }
+        }
+        if (pointSum[ref] == null) {
+            readjustPointSum(ref, point);
+        }
+        return pointSum[ref];
     }
 
-    /**
-     * creates the bounding box of a node/leaf assuming that caching is enabled
-     *
-     * @param nodeReference node in question
-     * @return the bounding box
-     */
-
-    AbstractBoundingBox<double[]> getBoundingBoxReflate(Integer nodeReference) {
-        if (isLeaf(nodeReference)) {
-            return new BoundingBox(getPointFromLeafNode(nodeReference));
+    @Override
+    void readjustPointSum(Integer node, double[] point) {
```

Comment:
Same as above, I kind of like "recompute" better than "readjust" for this.

Reply:
Done.